### PR TITLE
toggle user if org has been enabled/disabled

### LINKED ISSUE #57: On toggle org, associated users should also be toggled.
When organisations are made active/inactive, any associated users (both primary and secondary) should be made active/inactive too. See [this issue](https://github.com/emfoundation/ce100-app/issues/422) for more detail.

Need to update toggleActive for organisations to reflect the desired effect.

### PR DESCRIPTION
**Pull request by mk4111**
Closes #57 

Adds trigger and procedure to toggle any associated users when org has been enabled/disabled.
Adds tests

### CODE REVIEW

**Comment by mk4111** [comment]
Hi @SimonLab and @JMurphyWeb - if you could have a wee look at this PR, that'd be great! Thank you!

File: tests/orgs/toggleActive/endpoint.test.js [hunk 1]
```diff
@@ -4,76 +4,177 @@ var test = require('tape');
...
+    url: '/peopleGetById?id=' + id
+  }
+}
+
+var allUsers = {
+  method: 'GET',
+  url: '/people'
+}
+
+// test active org with active primary user
```
Comment (jay-meister) [inline tests/orgs/toggleActive/endpoint.test.js:33]: There is no test for active org with inactive primary user.

File: lib/fixtures/set-triggers.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+$BODY$
+LANGUAGE plpgsql;
+
+-- Create triggers
+CREATE TRIGGER toggleOrgUsers
+AFTER
+UPDATE ON organisations
+FOR EACH ROW
+WHEN (OLD.active IS DISTINCT FROM NEW.active)
+EXECUTE PROCEDURE toggleOrgUsers();
```
Comment (jay-meister) [inline lib/fixtures/set-triggers.sql:24]: From the sql function, it looks like an inactive primary user would become activated when we toggle the linked organisations activity status.. 

A quick work-around could be to call this function only when the org is being deactivated. Or perhaps we can pass the function a parameter which indicates if the org is being activated or deactivated?

Comment (jay-meister) [reply lib/fixtures/set-triggers.sql:24]: from this comment we should go with whatever is easiest: https://github.com/emfoundation/ce100-app/issues/509#issuecomment-269489413

I think the easiest is to make this function run only when the org is being deactivated.

File: tests/orgs/toggleActive/endpoint.test.js [hunk 3]
```diff
@@ -4,76 +4,177 @@ var test = require('tape');
...
+            'successful enabling of org returns an empty array'
+          );
+          server.inject(orgDetails(orgId), function (res) {
+            t.ok(
+              res.result.org.active,
+              'previously inactive org is now enabled'
+            );
+            server.inject(userDetails(userId), function (res) {
+              t.ok(
+                !res.result[0].active,
```
Comment (jay-meister) [inline tests/orgs/toggleActive/endpoint.test.js]: this suggests that the updated user is NOT active, which contradicts what is supposed to happen according to the comment.

**Comment by jay-meister** [comment]
@mk4111 @SimonLab I have simplified this so that the the function will not be run if the admin is re-enabling the organisation. It was the simplest solution as per Alex's suggestion

**Review by jay-meister — APPROVED** [review]
looks great!

**Comment by mk4111** [comment]
@SimonLab - could you please have a look at this? Thank you! :heart:

**Review by SimonLab — APPROVED** [review]
Looks good, thanks @mk4111 !

**Pull request merged**